src/main/java/br/com/brforgers/mods/disfabric/listeners/MinecraftEventListener.java

```diff
@@ -69,16 +71,26 @@ public static void init() {
             });
 
             ServerPlayConnectionEvents.JOIN.register((handler, $2, $3) -> {
-                if (DisFabric.config.announcePlayers && !DisFabric.stop) {
+                if (DisFabric.config.announcePlayers && !DisFabric.stop && !VanishService.isPlayerVanished(handler.player)) {
```

Comment:
This might need some anti-boilerplate later on, given this check is generally repeated. *More so a self-note.*